crates/bevy_asset/src/server/info.rs

```diff
@@ -450,7 +458,7 @@ impl AssetInfos {
                 RecursiveDependencyLoadState::Loaded
             }
             (_loading, 0) => RecursiveDependencyLoadState::Loading,
-            (_loading, _failed) => RecursiveDependencyLoadState::Failed,
+            (_loading, _failed) => RecursiveDependencyLoadState::Failed(rec_dep_error.unwrap()),
```

Comment:
To preserve the previous behavior, I separated the errors based on where they were found. I'm not sure this makes sense though. In general, the interaction between `DependencyLoadState` and `RecursiveDependencyLoadState` is pretty confusing.

IMO we should simplify to just `LoadState` and `DependencyLoadState` (which are affected by their own dependencies recursively) unless there is a clear use case for distinguishing between deps and recursive deps.

Reply:
Totally agree. Follow-up though :)